connections/sim/sim_connection_factory.py

```diff
@@ -20,18 +19,41 @@ def requiresComPort(self) -> bool:
     def requiresBaudRate(self) -> bool:
         return False
 
+    def _getParentDir(self, dir):
+        dirs = dir.split("/")
+        return "/".join(dirs[0:len(dirs) - 1])
```

Comment:
This will likely not work on Windows due to it using backslashes instead.

Maybe we can use this Python built-in solution: https://stackoverflow.com/questions/2860153/how-do-i-get-the-parent-directory-in-python

Reply:
It may just work, actually - Windows will let you use forward slashes and resolves them to backslashes.
That being said, the most Pythonic way of going about this I think would be through the high level `Pathlib` library which provides object oriented filesystem manipulation. (I've never gotten the hang of it though and usually still use lower level `os.path.join` etc., directly manipulating strings)

Reply:
The join will work. The split won't.

Reply:
Yeah it uses pathlib now